src/sensitivity_interface.jl

```diff
@@ -67,17 +69,90 @@ instead of just at discrete data points.
+\frac{d\lambda^{\star}}{dt}=g_{u}(u(t,p),p)-\lambda^{\star}(t)f_{u}(t,u(t,p),p),\thinspace\thinspace\thinspace\lambda^{\star}(T)=0
+```
+
+and obtaining the sensitivities through the integral:
+
+```math
+\frac{dG}{dp}=\int_{t_{0}}^{T}\lambda^{\star}(t)f_{p}(t)+g_{p}(t)dt+\lambda^{\star}(t_{0})u_{p}(t_{0})
+```
+
+As defined, that cost function only has non-zero values over nontrivial intervals. However, in many
+cases one may want to include in the cost function loss values at discrete points, for example, matching
+the data at time points `t`. In this case, terms of `g` can be represented by Dirac delta functions
+which are then applied in the corresponding ``\lambda^\star`` and ``\frac{dG}{dp}`` equations. 
+
+For more information, see [Sensitivity Math Details](@ref sensitivity_math).
+
+## Positional Arguments
+
+- `sol`: the solution from the forward pass of the ODE. Note that if not using a checkpointing
+  sensitivity algorithm, then it's assumed that the (dense) interpolation of the forward solution is
+  of sufficient accuracy for recreating the solution at any time point.
+- `alg`: the algorithm to use for the solution of the adjoint problem.
+
+## Keyword Arguments
+
+- `t`: the time points at which the discrete cost function is to be evaluated.
+  This argument is only required if discrete cost functions are declared.
+- `g`: the continuous instantanious cost ``g(u,p,t)`` at a given time point
+  represented by a Julia function `g(u,p,t)`. This argument is only required
+  if there is a continuous 
+- `dgdu_discrete`: the partial derivative ``g_u`` evaluated at the discrete
+  (Dirac delta) times. If discrete cost values are given, then `dgdu_discrete`
+  is required.
+- `dgdp_discrete`: the partial derivative ``g_p`` evaluated at the discrete
+  (Dirac delta) times. If discrete cost values are given, then `dgdp_discrete`
+  is not required ans is assumed to be zero.
+- `dgdu_continuous`: the partial derivative ``g_u`` evaluated at times
+  not corresponding to terms with an associated Dirac delta. If `g` is given,
+  then this term is not required and will be approximated by numerical or autoamatic
+  differentiation (via the `autojacvec` keyword argument in the `sensealg`)
+  if this term is not given by the user.
+- `dgdp_continuous`: the partial derivative ``g_p`` evaluated at times
+  not corresponding to terms with an associated Dirac delta. If `g` is given,
+  then this term is not required and will be approximated by numerical or autoamatic
+  differentiation (via the `autojacvec` keyword argument in the `sensealg`)
+  if this term is not given by the user.
+- `abstol`: the absolute tolerance of the adjoint solve. Defaults to `1e-3`
+- `reltol`: the relative tolerance of the adjoint solve. Defaults to `1e-3`
+- `checkpoints`: the values to use for the checkpoints of the reverse solve, if the
+  adjoint `sensealg` has `checkpointing = true`. Defaults to `sol.t`, i.e. the
+  saved points in the `sol`.
+- `corfunc_analytical`:
```

Comment:
this is for the shadow adjoints?

Reply:
it's the correction term stemming from the required conversion between Stratonovich and Ito sense, which can be helpful to avoid the additional nested differentiation that otherwise occurs:
https://github.com/SciML/SciMLSensitivity.jl/blob/14fb96ea03ef70b224a5cb7a63b09c04ac82ebd6/test/sde_scalar_ito.jl#L85
(Since one has to convert twice, it's 2 times the standard conversion rule term.)